server/src/inmantals/server.py

```diff
@@ -118,22 +125,83 @@ def flatten(self, line, char):
         assert char < 100000
         return line * 100000 + char
 
+    def create_tmp_project(self) -> str:
+        ten_random_digits = "".join(random.choice(string.digits) for _ in range(10))
+        project_name = "project" + ten_random_digits
+        self.project_dir = os.path.join("/tmp", "vs-code", project_name)
+
+        os.makedirs(self.project_dir, exist_ok=True)
+        logger.info(f"project created at {self.project_dir}")
+
+        os.mkdir(os.path.join(self.project_dir, "libs"))
+
+        install_mode = InstallMode.master
+
+        modulepath = ["libs", os.path.dirname(self.rootPath)]
+
+        with open(os.path.join(self.project_dir, "project.yml"), "w+") as fd:
+            metadata: typing.Mapping[str, object] = {
+                "name": "testcase",
+                "description": "Project for testcase",
+                "repo": yaml.safe_load(self.repos),
+                "modulepath": modulepath,
+                "downloadpath": "libs",
+                "install_mode": install_mode.value,
+            }
+            yaml.dump(metadata, fd)
+
+        v2_metadata_file: str = os.path.join(self.rootPath, ModuleV2.MODULE_FILE)
+        v1_metadata_file: str = os.path.join(self.rootPath, ModuleV1.MODULE_FILE)
+
+        module_name: Optional[str] = None
+        if os.path.exists(v2_metadata_file):
+            mv2 = ModuleV2(project=None, path=self.rootPath)
+            module_name = mv2.name
+        elif os.path.exists(v1_metadata_file):
+            mv1 = ModuleV1(project=None, path=self.rootPath)
+            module_name = mv1.name
+
+        if not module_name:
+            raise ModuleNotFoundError
```

Comment:
How does this get handled when there is no module and no project?

Reply:
Added a custom exception + handle it as an error so that no trace is shown: 
`2023-03-01 11:43:46,763 __main__                 INFO    Starting language server
2023-03-01 11:43:46,766 inmantals.server         INFO    project created at /tmp/vs-code/project6025286518
2023-03-01 11:43:46,767 inmantals.server         ERROR   The Inmanta extension only works on projects and modules. Please make sure the current workspace is a valid project (see https://docs.inmanta.com/inmanta-service-orchestrator/latest/model_developers/project_creation.html) or module (see https://docs.inmanta.com/inmanta-service-orchestrator/latest/model_developers/module_creation.html). 
`

Reply:
but is it reported to the end-user in some way?

Reply:
For now, the message above is only displayed in the `Inmanta Language Server` console output. I think I'll add a toast notification